tests/test_common.js

```diff
@@ -512,3 +512,27 @@ test('testing timers are deleted correctly', t => {
     t.end()
   })
 })
+
+test('correct node behavior', t => {
+  const scope = nock('http://example.test')
+    .get('/')
+    .reply()
+
+  const reqSpy = sinon.spy()
+  const origHttpReq = http.request
+
+  http.request = reqSpy
+
+  http.get('http://example.test', res => {
+    t.equal(res.statusCode, 200)
+
+    res.on('data', reqSpy)
+
+    res.on('end', () => {
+      t.equal(reqSpy.called, false)
+      scope.done()
+      http.request = origHttpReq
+      t.end()
```

Comment:
When I run this test _without_ the change, I get a TypeError. Is that the expected behavior?

Could you clarify how this test exercises the bug from #1836?

Reply:
I added `http.request` to the spy function and now the test is failing on assertion instead of TypeError. 

The test makes sure `http.request` is not called when using `http.get`. That fixes #1836